django-stubs/core/checks/registry.pyi

```diff
@@ -18,7 +19,9 @@ class Tags:
     translation: str = ...
     urls: str = ...
 
-_CheckCallable = Callable[..., Sequence[CheckMessage]]
+class _CheckCallable(Protocol):
+    def __call__(self, app_configs: Apps, **kwargs: Any) -> Sequence[CheckMessage]: ...
```

Comment:
Perhaps this [should return an `Iterable`](https://github.com/django/django/blob/bccfaa23864a3960ec5562d2f95d1238f1348ee6/django/core/checks/registry.py#L89)?

And `app_configs` is kwarg-only, and we could add `databases`... (and test a `database`-using check)

```suggestion
    def __call__(self, *, app_configs: Apps, databases: Optional[Iterable[str]] **kwargs: Any) -> Iterable[CheckMessage]: ...
```

Reply:
Sure.

As for `databases`, I think Sequence is better than Iterable. It has more methods in common with `list` and the documentation also specifies this to be a list.